blueocean-dashboard/src/main/js/components/records.jsx

```diff
@@ -23,7 +23,7 @@ export class PipelineRecord extends Record({
     weatherScore: 0,
 }) {
     isFolder() {
-        return this.numberOfFolders !== null;
+        return this._class === 'io.jenkins.blueocean.service.embedded.rest.PipelineFolderImpl';
```

Comment:
Using _class is way more explicit than the num/folders/branches stuff which I always felt was a little too loose, so 🐝  - long term maybe referring to some kind of capability array with fixed values is a fine idea too.